### PR TITLE
Add ES6 syntax to Challenge' solution

### PR DESCRIPTION
**Pull request by xJuggl3r**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

File: guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md]:
```suggestion
```js
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally/index.md]:
```suggestion
```js
```

**Review by thecodingaviator — DISMISSED** [review]

File: guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md
Comment (RandellDawson) [reply guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md]: @thecodingaviator It is not necessary to use `js`.  `javascript` should work with prism just fine.

**Comment by RandellDawson** [comment]
I see no reason to add an ES6 syntax version of the guide as there is no different algorithm being implemented here.  It is strictly a syntax change of the same solution.

**Comment by xJuggl3r** [comment]
I agree there is no different algo, but on the other side, ES6 syntax should have some incentive, hence the suggestion.

**Comment by RandellDawson** [comment]
Added solutions should have different logic and not just a change in syntax.  In these specific challenges, you can not even use arrow function syntax through because of the need to use `this` so it is a mix of ES6 and conventional function syntax.  It would be (maybe) different if you were replacing a for loop with the use of a higher order function like map or reduce, but that is not what is going on here.

**Comment by xJuggl3r** [comment]
Gotcha, thanks for the explanation.

**Comment by RandellDawson** [comment]
Let's leave it open for some other mods to voice their opinions.

File: guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md
Comment (thecodingaviator) [reply guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md]: Agreed but since we're adding it to the other file, I figured why not do it here too?

**Comment by moT01** [comment]
I'm not sure on these - Do we need more than one solution per challenges article? Part of the problem seems to be that we don't have a template to use for these articles. Some have a beginner/intermediate/advanced solution. Some just have one solution. Some have many annoying icons. There is very little consistency across the articles.

To be more specific on this PR - I don't care either way. The current articles are quite short and another solution doesn't clutter them up. But the added solutions are really just the same as the ones already there with a different syntax. 

Yea, after looking closer - I think the best play here is to either not merge this or to replace the current solution with these new ones like @RandellDawson said. 

So if you can replace the current solutions with your new ones @xJuggl3r - I think we could probably merge this - assuming the solutions work.  And if you're going to do that - you could probably remove some of those curly brackets when using implicit returns like that as they are not needed.

**Comment by RandellDawson** [comment]
@xJuggl3r Can you please implement @moT01's latest suggestions.  Just replace the existing solution with the changes mentioned.

Thank you.

File: guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md
Comment (QuincyLarson) [reply guide/english/certifications/javascript-algorithms-and-data-structures/object-oriented-programming/use-an-iife-to-create-a-module/index.md]: JS works so I'm going to leave this.

**Review by QuincyLarson — APPROVED** [review]
Looks great. Thanks for these contributions, @xJuggl3r

**Pull request merged**
